src/AvaloniaMath/Rendering/AvaloniaElementRenderer.cs

```diff
@@ -1,103 +1,108 @@
-            }
-
-            RenderElement(box, x, y);
-
-            for (var i = 0; i < scaledTransformations.Count; ++i)
-            {
-       //TODO         _drawingContext.Pop();
-            }
+            //TODO         _drawingContext.Pop();
         }
+    }
 
-        public void FinishRendering()
-        { }
+    public void FinishRendering()
+    { }
 
-        private void RenderBackground(Box box, double x, double y)
+    private void RenderBackground(Box box, double x, double y)
+    {
+        if (box.Background != null)
         {
-            if (box.Background != null)
-            {
-                // Fill background of box with color:
-                _foregroundContext.FillRectangle(
-                    ((AvaloniaBrush)box.Background).Get(),
-                    new Rect(_scale * x, _scale * (y - box.Height),
-                        _scale * box.TotalWidth,
-                        _scale * box.TotalHeight));
-            }
+            // Fill background of box with color:
+            _foregroundContext.FillRectangle(
+                box.Background.ToAvalonia(),
+                new Rect(_scale * x, _scale * (y - box.Height),
+                    _scale * box.TotalWidth,
+                    _scale * box.TotalHeight));
         }
+    }
 
-        private static Transform ToTransform(Transformation transformation)
+    private static Transform ToTransform(Transformation transformation)
+    {
+        switch (transformation.Kind)
         {
-            switch (transformation.Kind)
-            {
-                case TransformationKind.Translate:
-                    var tt = (Transformation.Translate) transformation;
-                    return new TranslateTransform(tt.X, tt.Y);
-                case TransformationKind.Rotate:
-                    var rt = (Transformation.Rotate) transformation;
-                    return new RotateTransform(rt.RotationDegrees);
-                default:
-                    throw new NotSupportedException($"Unknown {nameof(Transformation)} kind: {transformation.Kind}");
-            }
+            case TransformationKind.Translate:
+                var tt = (Transformation.Translate) transformation;
+                return new TranslateTransform(tt.X, tt.Y);
+            case TransformationKind.Rotate:
+                var rt = (Transformation.Rotate) transformation;
+                return new RotateTransform(rt.RotationDegrees);
+            default:
+                throw new NotSupportedException($"Unknown {nameof(Transformation)} kind: {transformation.Kind}");
         }
+    }
 
-        /// <summary>
-        /// Generates the guidelines for WPF render to snap the box boundaries onto the device pixel grid.
-        /// </summary>
+    /// <summary>
+    /// Generates the guidelines for WPF render to snap the box boundaries onto the device pixel grid.
+    /// </summary>
     /*    private GuidelineSet GenerateGuidelines(Box box, double x, double y) => new GuidelineSet
```

Comment:
Are comment above valid? Do you still need this function?

Reply:
Same as above, will be the subject of further review in https://github.com/ForNeVeR/wpf-math/pull/325.